DominoSwissWeatherstation/module.php

```diff
@@ -41,11 +40,37 @@ public function ReceiveData($JSONString) {
 		if ($data->Values->ID == $this->ReadPropertyInteger("ID")) {
 			switch ($data->Values->Command) {
 				case 32:
-					SetValue($this->GetIDForIdent("LightValue"), $this->GetLightValue(intval($data->Values->Value / 8), ($data->Values->Value % 8)));
+					$newStep = intval($data->Values->Value / 8);
+					$newValue = $this->GetSimpleLightValue($newStep);
+
+					$lightId = $this->GetIDForIdent("LightValue");
+					$oldValue = GetValue($lightId);
+					$oldStep = $this->GetStepFromLightValue($oldValue);
+					if ($oldStep == -1) {
+						IPS_LogMessage("Invalid light value", "Old: " . $oldValue);
+						break;
```

Comment:
Are we sure we want to `break` if the OLD step is -1?
I would recommend to proceed without the check and just write the new value to the variable if the NEW step is > -1
we can use `return` instead of `break` if we want to break the case and also directly return from the function.

Reply:
Cood catch. Usually the old value should always be a valid one unless it gets overwritten manually for some reason. But yes, in that case we want to write the new value.